run_tests.sh

```diff
@@ -1,23 +1,25 @@
+#!/bin/bash
+
 #
 # Simple convenience test runner for the herd book.
 #
 
 VENV="venv"
 
-cd $(dirname "$0")
+cd "$(dirname "$0")" || exit 1
 
 if [ ! -d "$VENV" ]
 then
   echo "Warning: Could not find virtual environment '$VENV'."
   echo "Creating virtual environment '$VENV'".
 
-  python3 -m venv "$VENV"
-  [[ "$?" != "0" ]] && "Error: Couldn't create virtual environment" && exit 1
+  python3 -m venv "$VENV" || "Error: Couldn't create virtual environment" && exit 1
```

Comment:
```
if ! python3 -m venv "$VENV"; then
    echo 'Error: Could not create virtual environment' >&2
    exit 1
fi
```

This would be unambiguous and also introduces the missing `echo`.

This is the only thing that _needs_ to be fixed.